src/aead/block.rs

```diff
@@ -78,14 +79,56 @@ impl Block {
 impl From<&'_ [u8; BLOCK_LEN]> for Block {
     #[inline]
     fn from(bytes: &[u8; BLOCK_LEN]) -> Self {
-        unsafe { core::mem::transmute_copy(bytes) }
+        Self {
+            subblocks: [
+                u64::from_le_bytes(
+                    bytes[..8]
+                        .try_into()
+                        .expect("error converting leading bytes"),
+                ),
+                u64::from_le_bytes(
+                    bytes[8..]
+                        .try_into()
+                        .expect("error converting trailing bytes"),
+                ),
+            ],
+        }
     }
 }
 
 impl From_<&'_ [u8; 2 * BLOCK_LEN]> for [Block; 2] {
     #[inline]
     fn from_(bytes: &[u8; 2 * BLOCK_LEN]) -> Self {
```

Comment:
My plan is to eliminate the use of `Block` within `Poly1305`. Then I'm planning to experiment with split the `Block` code into `gcm::Block` and `aes::Block` types, or similar, that each only implement what AES and GCM use.

The original motivation for creating the `Block` type the way it exists today no longer applies and it just makes the code worse. I.e. basically this whole file is now a mistake and I want it to go away.

If you want to submit some PRs towards that end, I would appreciate them. The smaller the better.

Reply:
> If you want to submit some PRs towards that end, I would appreciate them. The smaller the better.

Happy to work on improving things. Would need maybe some more of your design ideas, what you would like to see instead.